src/client/datascience/notebook/notebookEditor.ts

```diff
@@ -198,12 +204,22 @@ export class NotebookEditor implements INotebookEditor {
         const status = this.statusProvider.set(DataScience.interruptKernelStatus(), true, undefined, undefined);
 
         try {
-            await kernel.interruptAllCells(this.document);
-            status.dispose();
+            const result = await kernel.interrupt(this.document);
+            if (result === InterruptResult.TimedOut) {
```

Comment:
Removed event as we didn't need it anymore, now that the interrupt takes place in KernelExecution and that can communicate (return) the result directly.

Reply:
Does the command route here? That would be how this could work.

Reply:
Yes, the command ends up calling this code (`INotebookEditor.interruptKernel`).